### ISSUE TITLE
resolveSourceType does not distinguish by destinationType

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#45]
```
What steps will reproduce the problem?
Create a situation where a single source type is mapped to more than one 
destination type over a period of time.
If explicit mappings are not registered, then the resolved source type can be 
different depending on the order in which the auto-mappings are triggered.

What is the expected output? What do you see instead?
The same source type should be resolved regardless of the order in which 
auto-mappings are triggered.

Please use labels and text to provide additional information.


```

Original issue reported on code.google.com by `matt.deb...@gmail.com` on 5 Sep 2012 at 11:01

**Comment by GoogleCodeExporter** [comment]
```
Test case to follow...
```

Original comment by `matt.deb...@gmail.com` on 5 Sep 2012 at 11:02

**Comment by GoogleCodeExporter** [comment]
```
See attached test case; issue has since been corrected in 1.2.0 branch by 
removing 'resolvedTypes' from MapperFacadeImpl
```

Original comment by `matt.deb...@gmail.com` on 6 Sep 2012 at 4:32
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

Attachments:
- [Issue45TestCase.java](https://storage.googleapis.com/google-code-attachments/orika/issue-45/comment-2/Issue45TestCase.java)

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.2.0 release
```

Original comment by `matt.deb...@gmail.com` on 7 Sep 2012 at 9:56
- Changed state: **Fixed**
